packages/devtools_app/lib/src/screens/memory/panes/leaks/leaks_pane.dart

```diff
@@ -148,122 +143,184 @@ class _LeaksPaneState extends State<LeaksPane>
+            ],
+          );
+        }
 
-  Future<void> _forceGC() async {
-    _analysis.status.value = AnalysisStatus.Ongoing;
-    _analysis.message.value = 'Forcing full garbage collection...';
-    await _invokeMemoryLeakTrackingExtension(
-      <String, dynamic>{
-        // TODO(polina-c): reference the constant in Flutter
-        // https://github.com/flutter/devtools/issues/3951
-        'forceGC': 'true',
+        return Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            AnalysisStatusView(
+              controller: _controller.status,
+              analysisStarter: Row(
+                children: [
+                  const _InformationButton(),
+                  _AnalyzeButton(analysis: _controller.status),
+                  _ForceGCButton(analysis: _controller.status),
+                ],
+              ),
+            ),
+            Expanded(
+              child: SingleChildScrollView(
+                child: ValueListenableBuilder<String>(
+                  valueListenable: _controller.leakSummaryHistory,
+                  builder: (_, value, __) => Text(value),
+                ),
+              ),
+            ),
+          ],
+        );
       },
     );
-    _analysis.status.value = AnalysisStatus.ShowingResult;
-    _analysis.message.value = 'Full garbage collection initiated.';
   }
+}
 
-  Future<void> _invokeMemoryLeakTrackingExtension(
-    Map<String, dynamic> args,
-  ) async {
-    await serviceManager.service!.callServiceExtension(
-      memoryLeakTracking,
-      isolateId: serviceManager.isolateManager.mainIsolate.value!.id!,
-      args: args,
-    );
-  }
+class _InformationButton extends StatelessWidget {
+  const _InformationButton({Key? key}) : super(key: key);
 
   @override
   Widget build(BuildContext context) {
-    final informationButton = Tooltip(
+    return Tooltip(
       message: 'Open memory leak tracking guidance.',
       child: IconButton(
         icon: const Icon(Icons.help_outline),
         onPressed: () async => await launchUrl(linkToGuidance, context),
       ),
     );
+  }
+}
 
-    if (_leakSummaryHistory.isEmpty)
-      return Column(
-        children: [
-          informationButton,
-          const Text('No information about memory leaks yet.'),
-        ],
-      );
+class _AnalyzeButton extends StatelessWidget {
+  const _AnalyzeButton({Key? key, required this.analysis}) : super(key: key);
+
+  final AnalysisStatusController analysis;
 
-    final analyzeButton = Tooltip(
+  @override
+  Widget build(BuildContext context) {
+    return Tooltip(
       message: 'Analyze the leaks and download the result\n'
-          'to ${_file_prefix}_<time>.yaml.',
+          'to ${_Constants.file_prefix}_<time>.yaml.',
       child: MaterialButton(
         child: const Text('Analyze and Download'),
         onPressed: () async => _requestLeaks(),
       ),
     );
+  }
+
+  Future<void> _requestLeaks() async {
```

Comment:
move this business logic to AnalysisStatusController

Reply:
analysis_status.dart is not concerned about memory leaks. It can serve any process.
May be we will want to reuse it and move it to src/primitives  eventually.
_requestLeaks should belong to _LeaksPaneController that handles leak tracking.